### PR TITLE
V2.8.0 cherry picks

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [ ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [ ] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [ ] I've described my changes in the section below.

### Changes description
Branch collecting chosen changes for v2.8.0 release

### CODE REVIEW

File: src/snowflake/cli/plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py [hunk 1]
```diff
@@ -102,10 +91,16 @@ def _pdf_v2_to_v1(v2_definition: DefinitionV20) -> DefinitionV11:
...
         ] = app_package_definition.distribution
     if app_package_definition.meta and app_package_definition.meta.post_deploy:
-        pdfv1["native_app"]["package"]["scripts"] = [
-            _convert_v2_post_deploy_hook_to_v1_scripts(s)
-            for s in app_package_definition.meta.post_deploy
-        ]
+        pdfv1["native_app"]["package"][
+            "post_deploy"
+        ] = app_package_definition.meta.post_deploy
+    if app_package_definition.meta:
```
Comment (sfc-gh-melnacouzi) [inline src/snowflake/cli/plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py:97]: this part and below is not needed (role and warehouse)

File: tests/nativeapp/test_v2_to_v1.py [hunk 2]
```diff
@@ -125,9 +125,11 @@
                     "package": {
                         "name": "pkg_name",
                         "distribution": "external",
-                        "scripts": [
-                            "scripts/script1.sql",
-                            "scripts/script2.sql",
+                        "role": "pkg_role",
+                        "warehouse": "pkg_wh",
```
Comment (sfc-gh-melnacouzi) [inline tests/nativeapp/test_v2_to_v1.py:129]: these 2 lines can be removed.

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -18,8 +18,14 @@
...
 
 ## Deprecations
+  * Added deprecation warning for `native_app.package.scripts` in project definition file.
 
 ## New additions
+* Added support for project definition file defaults in templates
+* Added support for `native_app.package.post_deploy` scripts in project definition file.
+  * These scripts will execute whenever a Native App Package is created or updated.
+  * Currently only supports SQL scripts: `post_deploy: [{sql_script: script.sql}]`
+* Added support for project definition file defaults in templates
```
Comment (sfc-gh-melnacouzi) [inline RELEASE-NOTES.md:28]: this line can be removed - it's duplicate.

File: src/snowflake/cli/plugins/nativeapp/run_processor.py [hunk 4]
```diff
@@ -155,30 +150,6 @@ def _execute_sql_script(
         except ProgrammingError as err:
             generic_sql_error_handler(err)
 
-    def _execute_post_deploy_hooks(self):
```
Comment (sfc-gh-melnacouzi) [inline src/snowflake/cli/plugins/nativeapp/run_processor.py:158]: the function above (_execute_sql_script) can also be removed.

File: tests/api/utils/test_definition_rendering.py [hunk 5]
```diff
@@ -622,11 +676,371 @@ def test_os_env_and_override_envs_in_version_1():
...
+    assert env.get("pkg_reference") == expected_pkg_name
+
+
+@pytest.mark.parametrize(
+    "definition",
+    [
+        {
+            "definition_version": "1.1",
+            "native_app": {
+                "name": "myapp",
```
Comment (sfc-gh-melnacouzi) [inline tests/api/utils/test_definition_rendering.py:888]: tests from here below (All the _suffixing tests) can be removed - they're failing now right?

File: tests/test_data/projects/integration_templated_v2/package/001-shared.sql [hunk 6]
```diff
@@ -0,0 +1,5 @@
+-- package script (1/2)
```
Comment (sfc-gh-melnacouzi) [inline tests/test_data/projects/integration_templated_v2/package/001-shared.sql:1]: this file doesn't seem to be used.

File: tests/test_data/projects/integration_templated_v2/package/002-shared.sql [hunk 7]
```diff
@@ -0,0 +1,12 @@
+-- package script (2/2)
```
Comment (sfc-gh-melnacouzi) [inline tests/test_data/projects/integration_templated_v2/package/002-shared.sql:1]: this file doesn't seem to be used.

File: tests_integration/nativeapp/test_init_run.py [hunk 8]
```diff
@@ -420,94 +420,6 @@ def test_nativeapp_init_from_repo_with_single_template(
             single_template_repo.close()
 
 
-# Tests that application post-deploy scripts are executed by creating a post_deploy_log table and having each post-deploy script add a record to it
```
Comment (sfc-gh-melnacouzi) [inline tests_integration/nativeapp/test_init_run.py:423]: there are a couple of imports that can be removed form this file as well.

**Comment by sfc-gh-pczajka** [comment]
Fixed in https://github.com/snowflakedb/snowflake-cli/pull/1469

**Pull request closed without merging**